backend/src/ee/services/dynamic-secret/providers/templateUtils.ts

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable func-names */
+import handlebars from "handlebars";
+import RE2 from "re2";
+
+import { alphaNumericNanoId } from "@app/lib/nanoid";
+
+export const compileUsernameTemplate = ({
+  usernameTemplate,
+  randomUsername,
+  identityName,
+  unixTimestamp,
+  options
+}: {
+  usernameTemplate: string;
+  randomUsername: string;
+  identityName?: string;
+  unixTimestamp?: number;
+  options?: {
+    toUpperCase?: boolean;
+  };
+}): string => {
+  // Pre-process template to replace {{random-N}} patterns before compiling
+  let processedTemplate = usernameTemplate;
+  const randomPattern = /\{\{random-(\d+)\}\}/g;
+  let match;
+
+  // eslint-disable-next-line no-cond-assign
+  while ((match = randomPattern.exec(usernameTemplate)) !== null) {
+    const fullMatch = match[0];
+    const length = parseInt(match[1], 10);
+
+    if (length > 0 && length <= 100) {
+      const randomValue = alphaNumericNanoId(length);
+      processedTemplate = processedTemplate.replace(fullMatch, randomValue);
+    }
+  }
+
+  // Register replace helper
+  handlebars.registerHelper(
+    "replace",
+    function (text: string, searchValue: string, replaceValue: string, limit?: number) {
```

Comment:
I think we can add limit later and simplify this with replaceAll command for now.